### PR TITLE
Address TianoCore review comments for "Silicon/NXP: Add i.MX6 ACPI tables" Patch

### PR DESCRIPTION
**Pull request by christopherco**
This patch addresses upstream reviewer comments to the i.MX6Pkg ACPI tables patch.

Original TianoCore upstream review comments: https://lists.01.org/pipermail/edk2-devel/2018-December/034101.html

### CODE REVIEW

File: Silicon/NXP/iMX6Pkg/AcpiTables/Dsdt-Enet.asl [hunk 1]
```diff
@@ -16,6 +16,7 @@ Device (ENET)
 {
   Name (_HID, "NXP010D")
   Name (_UID, 0x0)
+  Name (_CCA, 1)  // ENET is cache coherent
```
Comment (jordanrh1) [inline Silicon/NXP/iMX6Pkg/AcpiTables/Dsdt-Enet.asl]: >Name (_CCA, 1)  // ENET is cache coherent [](start = 2, length = 41)

What does this mean - is it used by the miniport or NDIS?

Comment (jordanrh1) [reply Silicon/NXP/iMX6Pkg/AcpiTables/Dsdt-Enet.asl]: I think what this means, is if the device is cache coherent, a CPU that accesses memory which has just been written by the device does NOT need to invalidate it's cache for that memory region before accessing that memory. It is my understanding that ARM is not generally cache coherent, and the CPU must invalidate it's cache before accessing that memory. Are you certain that a cache invalidation is not required?

---
In reply to: [246500631](https://github.com/ms-iot/imx-edk2-platforms/pull/4#discussion_r246500631) [](ancestors = 246500631)

**Review by jordanrh1 — CHANGES_REQUESTED** [review]
Need to be certain that _CCA=1 is correct. _CCA=0 is the safer option, since the CPU will perform cache invalidations and flushes on memory it shares with devices.

It looks like the system queries this property but I'm not seeing it used in any device stacks. I suspect that testing both ways will work.

File: Silicon/NXP/iMX6Pkg/AcpiTables/Dbg2.aslc [hunk 2]
```diff
@@ -13,8 +13,7 @@
 **/
 
 #include <Platform.h>
-
-#include "iMX6.h"
```
Comment (neilsh-msft) [inline Silicon/NXP/iMX6Pkg/AcpiTables/Dbg2.aslc:17]: These should be sorted alphabetically?  Above you have <imx6.h> followed by <platform.h>, here you have it reversed.

**Comment by neilsh-msft** [comment]
For Jordan's _CCA comments, I know on the iMX8 platform I have to use _CCA set to zero for the SDHC controller.

**Comment by christopherco** [comment]
Regarding _CCA, I do agree _CCA=0 is the safer option.

~~My conundrum is that Windows default behavior is cache-coherent access. On x86/amd64, which is what the inbox driver stacks were initially targeted towards, accesses are cache coherent. _CCA was only recently added in ACPI 6.0+ to handle ARM's inconsistent cache-coherency policies. Since the Windows policy was initially based on x86/amd64 policy, the absence of _CCA implies _CCA=1.~~

~~That being said, I see no functional harm in changing _CCA to 0 for these. It would be a (probably minor) perf hit but it is the safer option.~~

**Comment by christopherco** [comment]
Correction: My above comment was only true for ARM64. HAL enforces that the absence of _CCA implies _CCA=0 on ARM32. Will change

File: Silicon/NXP/iMX6Pkg/AcpiTables/Dbg2.aslc (hunk 2 above)
Comment (christopherco) [reply Silicon/NXP/iMX6Pkg/AcpiTables/Dbg2.aslc:17]: This was an oversight. Will fix.

**Review by jordanrh1 — APPROVED** [review]

**Review by neilsh-msft — APPROVED** [review]

**Pull request merged**